--- mshowalign2.c	2021-05-27 16:22:21.496057292 +0200
+++ mshowalign2_patched.c	2021-05-27 16:25:01.181257520 +0200
@@ -745,9 +745,9 @@
 	if (info_str[0]) fprintf(fp,"; %s_info: %s\n",m_msp->f_id0,info_str);
 	if (ppst->zsflag>=0) 
 	  fprintf (fp,"; %s_z-score: %4.1f\n; %s_bits: %3.1f\n; %s_expect: %6.2g\n",
-		   m_msp->f_id0,bbp->zscore,
-		   m_msp->f_id0,zs_to_bit(bbp->zscore, m_msp->n0, bbp->seq->n1),
-		   m_msp->f_id0,bbp->rst.escore);
+		   m_msp->f_id0,lzscore,
+		   m_msp->f_id0,zs_to_bit(lzscore, m_msp->n0, bbp->seq->n1),
+		   m_msp->f_id0,lzscore);
 #else
 	if ((m_msp->markx & MX_M11OUT) == 0) {
 	  fprintf (fp,"; %s_%s: %d\n", m_msp->f_id0, m_msp->alab[0], lsw_score);
